### PR TITLE
Improvements on spring Jdbi repositories

### PR DESCRIPTION
**Pull request by xfredk**
- reduced use of reflection
- documented usage
- fixed minor bug

As English in not my first language, please check spelling and grammar.
Made some improvements as suggested by @stevenschlansker

(I fudged my rebase, so I decided to make a new branch instead)

### CODE REVIEW

File: docs/src/adoc/index.adoc [hunk 1]
```diff
@@ -7333,6 +7332,10 @@ To use this module, add a Maven dependency:
 </dependency>
 ----
 
+==== XML-based configuration
+For XML-based configurations the class `JdbiFactoryBean` is made available to set up a
```
Comment (hgschmie) [inline docs/src/adoc/index.adoc:7336]: could we use a link here as well?

File: docs/src/adoc/index.adoc [hunk 2]
```diff
@@ -7440,6 +7443,71 @@ Global defined attributes may be configured on the factory bean:
...
+    @Bean
+    public Jdbi jdbi(DataSource ds) {
+        ConnectionFactory cf = new SpringConnectionFactory(dataSource); // <1>
+        final Jdbi jdbi = Jdbi.create(cf);
+        /* additional configuration goes here */
+        return jdbi;
+    }
+}
+----
+<1> This connection factory will enable <<jta, JTA support>>.
```
Comment (hgschmie) [inline docs/src/adoc/index.adoc:7463]: I read this as an anchor to the general JTA documentation, not spring-jta. Could we make this `spring-jta` ?

File: docs/src/adoc/index.adoc [hunk 3]
```diff
@@ -7440,6 +7443,71 @@ Global defined attributes may be configured on the factory bean:
...
+
+[#jta]
+==== Synchronization between Jdbi and JTA
+If you use Spring and databases, you will most likely also use the Java Transaction API (JTA)
+to manage the transactions. The `JdbiUtil` utility can be used to obtain link:{jdbidocs}/core/Handle.html[Handles^]
+synchronized with Spring's JTA, as can be seen in the following fragment:
+
+[source,java,indent=0]
+----
+Handle handle = JdbiUtil.getHandle(jdbi); // <1>
```
Comment (hgschmie) [inline docs/src/adoc/index.adoc:7473]: Handle is a closeable object and the general guidance everywhere else in the doc is to either use callbacks (which does not apply here) or use try-with-resources. Could this be rewritten as

```
try (Handle handle = JdbiUtil.getHandle(jdbi)) { 
    /* use the hande */
} 
```

thus not needing the explicit "closeIfNeeded" call.

**Review by hgschmie — APPROVED** [review]
Added some minor comments, generally looks good. LMK if you want to address the comments, otherwise I will merge some time this week (this is a holiday week in the U.S.).

File: docs/src/adoc/index.adoc (hunk 3 above)
Comment (hgschmie) [reply docs/src/adoc/index.adoc:7473]: looking through the code, I think that is correct. I don't know enough about Spring to be able to make meaningful comments on the JdbiUtil class.

**Comment by hgschmie** [comment]
LMK when you are done; I am planning to do a 3.42.0 release (the Kotlin coroutine stuff and the spring improvements are sufficient for a new release) now that the tests are stable again.

**Review by stevenschlansker — APPROVED** [review]

**Pull request merged**